src/commcare_cloud/ansible/roles/commcarehq/templates/celery_task_kill.sh.j2

```diff
@@ -0,0 +1,19 @@
+#!/bin/bash 
+
+# Process to soft and hard kill old celery jobs.
+#
+
+HOURS=$1
+MAX_MINUTES=`expr ${HOURS} \* 60 \* 60`
+
+SIG_NAME=$2
+
+PIDS=$(ps -eo comm,pid,etimes | awk '/celeryd:/ { if ( $4 > '"${MAX_MINUTES}"' ) { print $3 } }');
+
+if [ "$PIDS" != "" ]; then
+  for pid in ${PIDS}
+  do
+     #echo "kill -${SIG_NAME} ${pid}"
```

Comment:
If you uncomment this, would it send us an email? I think that might be preferable if it does

Reply:
@dannyroberts, not completely sure. I have enabled it in the staging celery server will monitor for sometime and let you know. thanks

Reply:
Getting this in emails from staging:
```
kill -SIGTERM 8449
kill -SIGTERM 8453
kill -SIGTERM 9798
```

Would be great to get more detail:
```
if [ "$PIDS" != "" ]; then
  echo "Killing stale celery workers:
  echo "ETIME COMMAND
  for pid in ${PIDS}
  do
     echo `ps -o etime= -o args= $PID`
     ....
```

Reply:
@snopoke thanks for the suggestions. Will update the script and commit the changes.

Reply:
@snopoke @dannyroberts updated the template like this
```
if [ "$PIDS" != "" ]; then
  echo "Killing stale celery workers:"
  echo "Process information with start and elapsed time:"
  for pid in ${PIDS}
  do
     echo "`ps -p ${pid} -o pid,start,etime`"
     echo "kill -${SIG_NAME} ${pid}"
     kill -${SIG_NAME} ${pid}
  done
fi
```
sample output:
```
Killing stale celery workers:
Process information with start and elapsed time:
  PID  STARTED     ELAPSED
20852 14:06:37    01:31:25
kill -SIGTERM 20852
  PID  STARTED     ELAPSED
20854 14:06:38    01:31:24
kill -SIGTERM 20854
  PID  STARTED     ELAPSED
20856 14:06:38    01:31:24
kill -SIGTERM 20856
```